checksum.go

```diff
@@ -183,15 +184,15 @@ func newChecksumFromValue(checksumValue, filename string) (*FileChecksum, error)
 	return c, nil
 }
 
-// ChecksumFromFile will return all the FileChecksums found in file
+// ChecksumFromFile will return the first FileChecksums found in file.
 //
-// ChecksumFromFile will try to guess the hashing algorithm based on content
-// of checksum file
+// ChecksumFromFile will try to guess the hashing algorithm based on the
+// content of a checksum file.
 //
-// ChecksumFromFile will only return checksums for files that match file
-// behind src
-func (c *Client) ChecksumFromFile(checksumFile string, src *url.URL) (*FileChecksum, error) {
-	checksumFileURL, err := urlhelper.Parse(checksumFile)
+// ChecksumFromFile will only return checksums for files that match
+// checksummedURL, which is the object being checksummed.
+func (c *Client) ChecksumFromFile(ctx context.Context, checksumURL, checksummedURL string) (*FileChecksum, error) {
```

Comment:
actually, one more nitpick - now that we pass `u.Path`, `checksummedURL` should be `checksummedPath`?

Reply:
Ah I didn't see this one sorry, you are correct. I will make a new PR changing this.

Reply:
Ah I pushed it directly to `v2` 🙂